cheatsheets/JWT_Cheat_Sheet.md

```diff
@@ -0,0 +1,227 @@
+ signature(s) is acceptable. Most libraries have fixed this issue, but as a rule, if you're using a symmetric key you
+should explicitly set which libraries are allowed.
+
+Second (and not shown) is to use an asymmetric key to sign your jwts. In general, it should be seen as a best practice
+ to use an asymetrc key to sign jwts rather than a symmetric key.
+
+#### Implementation Example Symmetric Key
+
+=== "Java Example"
+    --8<-- "JWTCSA/0-verification.md:java"
+
+=== "Python Example (`pyjwt`)"
+    --8<-- "JWTCSA/0-verification.md:pyjwt"
+
+=== "JavaScript Example (`jose`)"
+    --8<-- "JWTCSA/0-verification.md:jose"
+
+### Token Sidejacking
+
+#### Symptom
+
+This attack occurs when a token has been intercepted/stolen by an attacker and they use it to gain access to the system
+ using targeted user identity. In part, this attack really can't be mitigated "in platform" as JWTs are generically stateless
+ by design.
+
+#### How to Prevent
+
+If you're using a jwt as a session token in the context of a webapp you should follow the
+ [Session Management Cheat Sheet](Session Management Cheat Sheet). But a more generic fix might be to use JWTs in conjunction
+ with OIDC/OAUTH2 and utilize the replay protections those meta libraries provide. However, if you're looking to implement
+ a similar solution you can utilize a `nonce`.
+
+Additionally, depending on your application and security designs; it might make sense to have your JWT tokens valid for
+ short periods of time; especially in a "service to service" context where the service can regenerate a token at any time.
+ There may be a performance concern associated with signing a jwt token on each request; but modern CPUs should be able to
+ generate tokens without much consideration for all but the most abnormal workloads. This wouldn't eliminate the problem
+ of token sidejacking, but it would limit the amount of time a successful sidejacking could be utilized.
+
+/// details | Stateful Considerations
+    type: warning
+
+Using a nonce in a multi-node environment will require state to be shared between each node. Generally this is done with
+ a database table or a tool like [redis](https://redis.io/). If you're application is designed to be stateless, or if it's
+ a microservices type architecture this approach may not be ideal, or might be somewhat difficult to utilize.
+///
+
+So in this example you're server would return a jwt with the `nonce` claim (along with the normal `nbf`,
+ `iat`, `exp` time based claims).
+
+/// details | ToDo: Nonce Recommendation
+    type: ToDo
+
+Find the "right way" to implement a nonce. Finding conflicting information about how it should work (just server verify,
+ just client verify, both verify); protections for simultaneous API calls etc...
+///
+
+### No Built-In Token Revocation by the User
+
+#### Symptom
+
+This problem is inherent to JWT because a token only becomes invalid when it expires. The
```

Comment:
Should we reference the [OAuth status List](https://www.ietf.org/archive/id/draft-ietf-oauth-status-list-00.html) draft for revocation-list based approach?